util/dvsim/SlurmLauncher.py

```diff
@@ -91,13 +94,18 @@ def poll(self):
 
         # Copy slurm job results to log file
         if os.path.exists(self.slurm_log_file):
-            with open(self.slurm_log_file, 'r') as slurm_file:
-                lines = slurm_file.readlines()
-                with open(self.deploy.get_log_path(), 'a') as out_file:
-                    for line in lines:
-                        out_file.write(line)
-                    out_file.flush()
-            os.remove(self.slurm_log_file)
+            try:
+                with open(self.slurm_log_file, 'r') as slurm_file:
+                    try:
+                        with open(self.deploy.get_log_path(), 'a') as out_file:
+                            shutil.copyfileobj(slurm_file, out_file)
+                    except IOError as e:
+                        raise LauncherError(f'File Error: {e} when handling '
+                                            f'{self.deploy.get_log_path()}')
+                # Remove the temporary file from the slurm process
+                os.remove(self.slurm_log_file)
```

Comment:
Isn't this essentially just moving the file? I don't think we need to open either copy, do we? Can we just use `shutil.move`?

Reply:
No it's not. It is appending the log files from all slurm jobs to a single log file.

Reply:
Ah, I see! Cool, that makes sense then :-)

Reply:
Oh, wait a sec! Does that mean we're doing something like:
```
cp acc a; rm a
cp acc b; rm b
cp acc c; rm c
```

Maybe your original code made sense (essentially doing a `cat`)?

Reply:
Yep, we are doing that. But shouldn't that work via shutil. copyfileobj()?